dali/pipeline/data/views.h

```diff
@@ -181,16 +164,17 @@ view(const TensorVector<Backend> &data) {
   if (data.size() == 0)
     return {};
   using U = std::remove_const_t<T>;
-  detail::enforce_dim_in_view<ndim>(data.shape());
-  TensorListView<detail::storage_tag_map_t<Backend>, T, ndim> ret;
-  ret.shape = convert_dim<ndim>(data.shape());
-  ret.data.resize(ret.shape.num_samples());
-  for (int i = 0; i < ret.shape.num_samples(); i++) {
-    ret.data[i] = data[i].template data<U>();
+  const auto &shape = data.shape();
+  detail::enforce_dim_in_view<ndim>(shape);
+
+  std::vector<T *> ptrs(shape.num_samples());
+  for (int i = 0; i < shape.num_samples(); i++) {
+    ptrs[i] = data[i].template data<U>();
   }
-  return ret;
+  return { std::move(ptrs), convert_dim<ndim>(shape) };
```

Comment:
Out of sheer curiosity, I've checked - this variant is over 10% faster. Good job!
:+1:

Reply:
It might be due to the fact that we do not copy the shape twice?